File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.java
class Solution {
    public int solution(int n, int[] stations, int w) {
        int range = w*2+1;
        int answer = 0;
        int e=0;
        for(int a : stations) {
            if(a-w-1-e > 0) {
                answer += (a-w-1-e+range-1)/range;
            }
            e = a+w;
        }
        if(e < n) {
            answer += (n-e+range-1)/range;
        }

        return answer;
    }
}

File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.cpp
#include <iostream>
#include <vector>
using namespace std;

int solution(int n, vector<int> stations, int w)
{
    int answer = 0;

    int s = 0, e = 0;
    int dist = w*2 + 1;
    for(int station : stations){
        s = station - w;
        
        if(s < 0) s = 0;
        answer += (s-e-1+dist-1)/dist;
        e = station + w;
    }
    
    if(e < n) {
        answer += (n-e+dist-1)/dist;
    }

    return answer;
}